cloudconfig/containerinit/container_userdata_test.go

```diff
@@ -84,80 +96,181 @@ func (s *UserDataSuite) SetUpTest(c *gc.C) {
 		ConfigType:    network.ConfigManual,
 		NoAutoStart:   true,
 	}}
-	s.expectedSampleConfig = `
-auto any0 any1 any3 lo
-
-iface lo inet loopback
-  dns-nameservers ns1.invalid ns2.invalid
-  dns-search bar foo
 
-iface any0 inet static
-  address 0.1.2.3/24
-  gateway 0.1.2.1
-
-iface any1 inet static
-  address 0.2.2.4/24
-  post-up ip route add 0.5.6.0/24 via 0.2.2.1 metric 50
-  pre-down ip route del 0.5.6.0/24 via 0.2.2.1 metric 50
-
-iface any2 inet dhcp
-
-iface any3 inet dhcp
+	for _, version := range []string{
+		"/usr/bin/python2",
+		"/usr/bin/python3",
+		"/usr/bin/python",
+	} {
+		if _, err := os.Stat(version); err == nil {
+			s.pythonVersions = append(s.pythonVersions, version)
+		}
+	}
 
-iface any4 inet manual
-`
-	s.expectedSampleUserData = `
-#cloud-config
+	s.expectedSampleConfigWriting = `#cloud-config
 bootcmd:
 - install -D -m 644 /dev/null '%[1]s'
 - |-
   printf '%%s\n' '
-  auto any0 any1 any3 lo
+  auto lo {ethaa_bb_cc_dd_ee_f0} {ethaa_bb_cc_dd_ee_f1} {eth}
 
   iface lo inet loopback
     dns-nameservers ns1.invalid ns2.invalid
     dns-search bar foo
 
-  iface any0 inet static
+  iface {ethaa_bb_cc_dd_ee_f0} inet static
     address 0.1.2.3/24
     gateway 0.1.2.1
 
-  iface any1 inet static
+  iface {ethaa_bb_cc_dd_ee_f1} inet static
     address 0.2.2.4/24
     post-up ip route add 0.5.6.0/24 via 0.2.2.1 metric 50
     pre-down ip route del 0.5.6.0/24 via 0.2.2.1 metric 50
 
-  iface any2 inet dhcp
+  iface {eth} inet dhcp
 
-  iface any3 inet dhcp
+  iface {eth} inet dhcp
 
-  iface any4 inet manual
+  iface {eth} inet dhcp
   ' > '%[1]s'
-runcmd:
+`
+	s.expectedSampleConfig = `
+auto lo {ethaa_bb_cc_dd_ee_f0} {ethaa_bb_cc_dd_ee_f1} {eth}
+
+iface lo inet loopback
+  dns-nameservers ns1.invalid ns2.invalid
+  dns-search bar foo
+
+iface {ethaa_bb_cc_dd_ee_f0} inet static
+  address 0.1.2.3/24
+  gateway 0.1.2.1
+
+iface {ethaa_bb_cc_dd_ee_f1} inet static
+  address 0.2.2.4/24
+  post-up ip route add 0.5.6.0/24 via 0.2.2.1 metric 50
+  pre-down ip route del 0.5.6.0/24 via 0.2.2.1 metric 50
+
+iface {eth} inet dhcp
+
+iface {eth} inet dhcp
+
+iface {eth} inet dhcp
+`
+	s.expectedSampleUserData = `
```

Comment:
This makes me wonder if we want to try to be terser with our arguments. I think some clouds have limits on the number of characters for user-data (64k, or so).
I think we compress the content, so maybe it doesn't matter.